tests/product/timing_test_decorator.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from re import match
+from time import time
+
+
+def _initialize_module_logger(module_name):
+    logger = logging.getLogger(module_name)
+    logger.setLevel(logging.INFO)
+
+    message_format = '%(levelname)s %(name)s - %(message)s'
+    formatter = logging.Formatter(message_format)
+
+    info_file_handler = logging.FileHandler(filename='product_test_info.log',
+                                            mode='a')
+    info_file_handler.setLevel(logging.INFO)
+    info_file_handler.setFormatter(formatter)
+
+    error_file_handler = logging.FileHandler(filename='product_test_error.log',
+                                             mode='a')
+    error_file_handler.setLevel(logging.ERROR)
+    error_file_handler.setFormatter(formatter)
+
+    if not len(logger.handlers):
+        logger.addHandler(info_file_handler)
+        logger.addHandler(error_file_handler)
+    return logger
+
+
+def _name_starts_with_test(name):
+    pattern = '^test.*$'
+    return match(pattern, name)
+
+
+def log_function_time(module_name="root"):
+    """
+    Logs the execution time of the input function
+    """
+    def name_wrapper(some_function):
+        def time_wrapper(*args, **kwargs):
+            logger = _initialize_module_logger(module_name)
+            function_name = some_function.__name__
+            function_is_a_test = _name_starts_with_test(function_name)
+            if function_is_a_test:
+                logger.info("%s started", function_name)
+            start_time = time()
+
+            return_value = some_function(*args, **kwargs)
```

Comment:
well i don't catch them, so they will just bubble up the stack. is that a problem?